sockeye/train.py

```diff
@@ -604,7 +607,7 @@ def create_model_config(args: argparse.Namespace,
     config_decoder = create_decoder_config(args, encoder_num_hidden)
 
     source_factor_configs = None
-    if config_data.num_source_factors > 1:
+    if len(source_vocab_sizes) > 1:
```

Comment:
it probably makes sense to add a check for the args.source_factors_num_embed being of equal size to source_factor_vocab_sizes.